test-data/unit/check-async-await.test

```diff
@@ -1021,3 +1021,19 @@ def coro() -> Generator[int, None, None]:
 reveal_type(coro)  # N: Revealed type is "def () -> typing.AwaitableGenerator[builtins.int, None, None, typing.Generator[builtins.int, None, None]]"
 [builtins fixtures/async_await.pyi]
 [typing fixtures/typing-async.pyi]
+
+[case asyncIteratorInProtocol]
+from typing import AsyncIterator, Protocol
+
+class P(Protocol):
+    async def launch(self) -> AsyncIterator[int]:
+        raise BaseException
+
+class Launcher(P):
+    def launch(self) -> AsyncIterator[int]:  # E: Return type "AsyncIterator[int]" of "launch" incompatible with return type "Coroutine[Any, Any, AsyncIterator[int]]" in supertype "P" \
+                                             # N: Consider declaring "launch" in supertype "P" without "async" \
```

Comment:
Shouldn't it instead say: "consider adding `async` to `def lauch()`"?

Because in my experience, AsyncIterators are way more common with `async def` rather than regular `def`

Reply:
Ehh, this is confusing :|

The 'problem' here should really be reported on this line:
```
  class P(Protocol):
!>    async def launch(self) -> AsyncIterator[int]:
         raise BaseException
```
but we don't put notices proactively on declarations that "look wrong", or do we?

The error on the derived class gives us the excuse to make a notice. What we want to say is "listen, we must error on `Launcher`'s method, but `Launcher`'s method is probably fine, consider changing the base declaration instead".